### ISSUE TITLE
Fix broken BSC swaps

### ISSUE THREAD

**Issue by onyb** [brave/brave-browser#25696]
|Post|https://community.brave.com/t/error-during-swaps-invalid-sender-bsc-network|
-|-

BSC swaps are broken due to incorrect assumption that the network supports EIP-1559. This is likely due to a recent upgrade in BSC nodes supporting `eth_feeHistory` even though it doesn't support Type-2 transactions.

#### BSC added a dummy `eth_feeHistory` implementation

```
$ curl -s https://bsc-dataseed1.binance.org/ -X POST -H "Content-Type: application/json" -d '{"id":1,"jsonrpc":"2.0","method":"eth_feeHistory","params":["0x28", "latest", [20,50,80]]}' | jq
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "oldestBlock": "0x14be044",
    "reward": [
      [
        "0x12a05f200",
        "0x12a05f200",
        "0x12d00e280"
      ]
    ],
    "baseFeePerGas": [
      "0x0",
      "0x0"
    ],
    "gasUsedRatio": [
      0.07355065963260536
    ]
  }
}
```

#### No `baseFeePerGas` in block headers though

```
$ curl -s https://bsc-dataseed1.binance.org -X POST -H "Content-Type: application/json" -d '{"id":1,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["latest", false]}' | jq '.result.baseFeePerGas'
# empty
```

**Comment by kjozwiak** [comment]
The above requires `1.44.104` or higher for `1.44.x` verification 👍

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.44.105 Chromium: 106.0.5249.91 (Official Build) (64-bit)
-- | --
Revision | `fa96d5f07b1177d1bf5009f647a5b8c629762157-refs/branch-heads/5249@{#707}`
OS | Windows 11 Version 22H2 (Build 22623.730)

- Verified steps from issue description
- Repro'd issue on older build and verified the fix after updating to `1.44.105`
- Verified able to submit a BNB Swap transaction

https://user-images.githubusercontent.com/17010094/193378369-1d2ce4c4-e274-446e-8806-40a28e493a3f.mov
